src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java

```diff
@@ -295,6 +296,12 @@ private AbstractReloadingMetadataResolver createMetadataResolver(final Settings
             AccessController.doPrivileged(new PrivilegedExceptionAction<Void>() {
                 @Override
                 public Void run() throws ComponentInitializationException {
+                    metadataResolver.setId(HTTPSamlAuthenticator.class.getName() + "_" + (++resolverIdCounter));
```

Comment:
Is it necessary to run this code with elevated privileges? Can it be limited to `metadataResolver.initialize();`?